Makefile

```diff
@@ -225,7 +227,7 @@ compile_as = $(CMD_PREFIX)mkdir -p `dirname $(1)`; \
 	     $(AS) $(ASFLAGS) $(call dynamic_flags,$(1),$(2)) -c $(2) -o $(1)
 compile_ld = $(CMD_PREFIX)mkdir -p `dirname $(1)`; \
 	     echo " LD        $(subst $(build_dir)/,,$(1))"; \
-	     $(CC) $(3) $(LDFLAGS) -Wl,-T$(2) -o $(1)
+	     $(CC) $(CFLAGS) $(3) $(LDFLAGS) -Wl,-T$(2) -o $(1)
 compile_ar = $(CMD_PREFIX)mkdir -p `dirname $(1)`; \
```

Comment:
Instead of using $(CFLAGS) for creating ELFs, we can add following flags to LDFLAGS:

LDFLAGS		+=	-Wl,-belf$(PLATFORM_RISCV_XLEN)-littleriscv
LDFLAGS		+=	-Wl,-melf$(PLATFORM_RISCV_XLEN)lriscv_$(PLATFORM_RISCV_ABI)

Reply:
Since you are using CC instead of LD directly, I think the march / mabi approach is more appropriate (plus there is the issue above regarding lp64d/ilp32d that will complicate things for no reason).

Reply:
Makes sense but this means only "-mabi", "-march", and "-mcmodel" are required in $(LDFLAGS). Better we append only these options to $(LDFLAGS) instead of all options in $(CFLAGS).

Reply:
-mabi is not understood by ld though, it'll be misleading to have it on LDFLAGS, it's not a big deal to pass CFLAGS to gcc, it's cleaner this way and more consistent IMHO.

Reply:
No problem, we can fix compile_ld() function in a different patch. Let's go with your change.

One last thing, first line of commit description is treated as subject of commit. This is "Make sure ld is always aware of the target abi/emulation" right now. Please update it to "Makefile: Make sure ld is always aware of the target abi/emulation" so that it is consistent with our commit description style. Same commit description style is followed in Linux and U-Boot as well.

My apologies for being picky about commit description style. We are trying to follow Linux coding style here so that developers don't have to adapt to new coding style.

Reply:
ACK I hope it's ok now ;-)